### ISSUE TITLE
Lighter pakage.json dependencies (remove puppeteer by default)

### ISSUE THREAD

**Issue by munrocket** [mrdoob/three.js#19332]
##### Description of the problem

Some of the developers cloning three.js repo without the intention to contribute. This is happening because they want to make tree-shake library or modify core for your own purposes in projects.

The idea is to introduce to two additional commands and suggest in console to install puppeteer only those developer who want to take screenshots in PRs.
```
"make-install": "npm i -D puppeteer@2.1.1 pngjs pixelmatch image-output serve-handler failonlyreporter",
"make-clean": "npm un puppeteer pngjs pixelmatch image-output serve-handler failonlyreporter",
```
But by doing so we modifying `package.json` after `npm run make-install` and facing another problem: we need to say to contributors not to add `package.json` or `package-lock.json`, which is relatively easy but little bit uncomfortable. I wonder if exist any safe way to keep `package.json` the same after install.

This idea is consistent with #19321 #19326

**Comment by donmccurdy** [comment]
An option would be to put a `package.json` for the screenshot tool into a subdirectory. Users who need to run it can run `npm install` in that directory; others would not. Dependencies from the parent folder are automatically available in the subdirectory.

**Comment by munrocket** [comment]
The solution was simple. Need to rethink this, +150mb is annoying.

**Comment by munrocket** [comment]
Ty.

**Comment by munrocket** [comment]
Solved, sanks.

**Comment by zlf2002171** [comment]
unsubscribe

2020-05-18 

zlf2002171 



发件人：munrocket <notifications@github.com>
发送时间：2020-05-11 21:47
主题：Re: [mrdoob/three.js] Lighter pakage.json dependencies (remove puppeteer by default) (#19332)
收件人："mrdoob/three.js"<three.js@noreply.github.com>
抄送："Subscribed"<subscribed@noreply.github.com>

Closed #19332.
